Cargo.toml

```diff
@@ -54,6 +56,7 @@ criterion = "0.3"
 gumdrop = "0.8"
 
 [features]
+no-threadpool = ["crossbeam-utils"]
```

Comment:
Rust features should be additive (see [here](https://doc.rust-lang.org/cargo/reference/features.html#feature-unification) for details), so I think the correct way to implement this is to have two additive features, and either require that only one of them can be active at a time, or define which one "wins". Maybe also a third "best" feature flag that uses a compile-time or runtime strategy for choosing the best option.

Reply:
I considered that, but using the rayon threadpool is going to be the best strategy in the vast majority of cases, so this is imposing complexity on the crate users for little gain. I don't see any practical problem *in this situation* with features being additive: if any dependent crate enables no-threadpool, then you don't get the threadpool, which seems fine to me. (You can think of it as enabling the use of crossbeam if you prefer.)

If we could detect WSL at compile time, I'd do so. But what we really want to know is whether the target executable is going to be run under WSL.

Reply:
Prioritizing other things right now, but it looks like detecting WSL dynamically can be made to work.

Reply:
I removed the `no-threadpool` feature; using rayon is unconditional for now. I've filed #315 to address the regression on WSL2.